eng/Directory.Build.Common.targets

```diff
@@ -196,6 +196,7 @@
     <Compile Include="$(AzureCoreSharedSources)ChangeTrackingDictionary.cs" LinkBase="Shared/Core" />
     <Compile Include="$(AzureCoreSharedSources)ChangeTrackingList.cs" LinkBase="Shared/Core" />
     <Compile Include="$(AzureCoreSharedSources)ClientDiagnostics.cs" LinkBase="Shared/Core" />
+		<Compile Include="$(AzureCoreSharedSources)TrimmingAttribute.cs" LinkBase="Shared/Core" />
```

Comment:
How widely is this type needed?  We might consider adding the shared source reference to the projects that need it directly, rather than putting it in the `<IncludeOperationSharedSource>` block.  Today, I think there are ~90 projects that use this MSBuild property, and take a code coverage hit by including shared source they don't use.

Reply:
This type isn't needed in any projects individually, but `DiagnosticScope.cs` and `DiagnosticScopeFactory.cs` use it. Since those are included in the operation shared source then this file has to as well so that the project can build.

There may be a better way to get around it that I'm not aware of, but I had looked at some older PR's related to DiagnosticScope and I think this is the approach needed.

Reply:
Now that I've gotten to the end of the PR, it looks like you may have already added the shared source reference to the relevant .csproj files, so you may not need the modification to Directory.Build.Common.targets anymore.  If you do still need it, I'm interested to understand this better.  Is it because of the DiagnosticScope dependency?

Reply:
Ah, you responded before I reloaded the page :)  Thank you, that makes sense!